x-pack/plugins/transform/public/app/hooks/use_api.ts

```diff
@@ -35,60 +56,69 @@ export const useApi = () => {
 
   return useMemo(
     () => ({
-      getTransforms(transformId?: TransformId): Promise<any> {
-        const transformIdString = transformId !== undefined ? `/${transformId}` : '';
-        return http.get(`${API_BASE_PATH}transforms${transformIdString}`);
+      getTransform(
+        transformId: TransformId
+      ): Promise<GetTransformsResponseSchema | HttpFetchError> {
+        return http.get(`${API_BASE_PATH}transforms/${transformId}`);
```

Comment:
this has confused me, will these not throw an error if the request fails, rather than resolve a `HttpFetchError` ?

Reply:
You're right of course, this code had issues! I wanted to use the approach described in the examples sections here: https://github.com/elastic/kibana/blob/master/examples/routing_example/public/services.ts

I missed changing the API to satisfy the specified types. I updated all related code in 8b22029. This introduced a bit of code churn but I decided to go for it in this PR:

- The transform's code will be in line with the example provided reference code now.
- I like the idea that only the API service/hook takes care of try/catch and the consuming code only asserts the resp type and doesn't try/catch. This enforces that we only catch errors returned from the API and make sure our consuming code doesn't trigger errors.